spec/unit/provider/acl/windows_spec.rb

```diff
@@ -125,440 +125,450 @@ def test_should_not_set_autorequired_user(user_name)
     end
 
     context '.get_ace_child_types' do
       it 'returns all if ace is nil' do
-        ace.stubs(:container_inherit?).returns(false) # ensure no false readings
-        ace.expects(:nil?).returns(true)
+        allow(ace).to receive(:container_inherit?).and_return(false) # ensure no false readings
+        expect(ace).to receive(:nil?).and_return(true)
 
-        Puppet::Provider::Acl::Windows::Base.get_ace_child_types(ace).must == :all
+        expect(Puppet::Provider::Acl::Windows::Base.get_ace_child_types(ace)).to eq :all
       end
 
       it 'returns none when container_inherit and object_inherit are both false' do
-        ace.expects(:container_inherit?).returns(false).at_least_once
-        ace.expects(:object_inherit?).returns(false).at_least_once
+        expect(ace).to receive(:container_inherit?).and_return(false).at_least(:once)
+        expect(ace).to receive(:object_inherit?).and_return(false).at_least(:once)
 
-        Puppet::Provider::Acl::Windows::Base.get_ace_child_types(ace).must == :none
+        expect(Puppet::Provider::Acl::Windows::Base.get_ace_child_types(ace)).to eq :none
       end
 
       it 'returns objects when container_inherit is false and object_inherit is true' do
-        ace.expects(:container_inherit?).returns(false).at_least_once
-        ace.expects(:object_inherit?).returns(true).at_least_once
+        expect(ace).to receive(:container_inherit?).and_return(false).at_least(:once)
+        expect(ace).to receive(:object_inherit?).and_return(true).at_least(:once)
 
-        Puppet::Provider::Acl::Windows::Base.get_ace_child_types(ace).must == :objects
+        expect(Puppet::Provider::Acl::Windows::Base.get_ace_child_types(ace)).to eq :objects
       end
 
       it 'returns containers when container_inherit is true and object_inherit is false' do
-        ace.expects(:container_inherit?).returns(true).at_least_once
-        ace.expects(:object_inherit?).returns(false).at_least_once
+        expect(ace).to receive(:container_inherit?).and_return(true).at_least(:once)
+        expect(ace).to receive(:object_inherit?).and_return(false).at_least(:once)
 
-        Puppet::Provider::Acl::Windows::Base.get_ace_child_types(ace).must == :containers
+        expect(Puppet::Provider::Acl::Windows::Base.get_ace_child_types(ace)).to eq :containers
       end
 
       it 'returns all when container_inherit and object_inherit are both true' do
-        ace.expects(:container_inherit?).returns(true).at_least_once
-        ace.expects(:object_inherit?).returns(true).at_least_once
+        expect(ace).to receive(:container_inherit?).and_return(true).at_least(:once)
+        expect(ace).to receive(:object_inherit?).and_return(true).at_least(:once)
 
-        Puppet::Provider::Acl::Windows::Base.get_ace_child_types(ace).must == :all
+        expect(Puppet::Provider::Acl::Windows::Base.get_ace_child_types(ace)).to eq :all
       end
     end
 
     context '.get_ace_propagation' do
-      before :each do
```

Comment:
why was the each block removed ?

Reply:
@TP `before` blocks should contain `expects` assertions according to rubocop.

Reply:
does that make it an allow, as we are stubbing ?

Reply:
I believe with the expect it is effectively stubbing but also asserting that the stub must be called  at some stage else it will fail the test.